pennylane/transforms/__init__.py

```diff
@@ -12,46 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """
-This subpackage contains QNode, quantum function, device, and tape transforms.
+This subpackage contains PennyLane transforms and their building blocks.
 
 
 .. currentmodule:: pennylane
 
-Transforms
-----------
+Custom transforms
+-----------------
 
-Transforms that act on QNodes
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+:class:`pennylane.transform` can be used to define custom transformations
+that work with PennyLane QNodes; such transformations can map a circuit
+to one or many new circuits alongside associated classical post-processing.
 
-These transforms accept QNodes, and return new transformed functions
-that compute the desired quantity.
+See the :class:`pennylane.transform` documentation for more details and examples.
 
 .. autosummary::
     :toctree: api
 
-    ~transforms.classical_jacobian
-    ~batch_params
-    ~batch_input
-    ~batch_partial
-    ~metric_tensor
-    ~adjoint_metric_tensor
-    ~specs
-    ~transforms.split_non_commuting
+    ~transforms.core.transform
 
-Transforms that act on quantum functions
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Transforms developer functions
+------------------------------
 
-These transforms accept quantum functions (Python functions
-containing quantum operations) that are used to construct QNodes.
+In this section of transform you can find the building blocks of transforms in PennyLane. The :class:`TransformDispatcher`
+and  :class:`TransformProgram` are developer-facing objects, that allows the creation, dispatching and composability
+of transforms.
 
 .. autosummary::
     :toctree: api
 
-    ~transforms.cond
-    ~defer_measurements
-    ~apply_controlled_Q
-    ~quantum_monte_carlo
-    ~transforms.insert
+    ~transforms.core.transform_dispatcher
+    ~transforms.core.transform_program
 
 Transforms for circuit compilation
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
```suggestion
Transforms library
------------------

A range of ready-to-use transforms are available in PennyLane.

Transforms for circuit compilation
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Also on line 50 please could we have:
> A set of transforms to perform basic circuit compilation tasks.

I think it makes less sense now to talk about what the transforms accept.

Reply:
Yes, updated!